src/presentation/controllers/device-controller.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseController, GetRequest, PostRequest, RestController } from './base-controller'
+import { Request, Response } from 'express'
+import { DeviceService } from '../../business-logic/services/interfaces/device-service'
+import { HttpErrorHandler } from '../interfaces/http-error-handler'
+import { Exception } from '../../model/exceptions/exception'
+import { HttpResponseFormatter } from '../interfaces/http-response-formatter'
+import { Device, INewsDevice, TelemetricsDevice } from '../../model/entities/device'
+import { DeviceType } from '../../model/enums/device-type'
+import { DeviceDtoInterface, INewsDeviceDto, TelemetricsDeviceDto } from '../dtos/device-dto'
+import { UnprocessableEntityException } from '../../model/exceptions/unprocessable-entity-exception'
+
+
+@RestController('/devices')
+export class DeviceController extends BaseController{
+
+  constructor(
+    private readonly deviceService: DeviceService,
+    private readonly httpErrorHandler: HttpErrorHandler,
+    private readonly httpResponseFormatter: HttpResponseFormatter,
+  ) {
+    super()
+  }
+
+  @GetRequest()
+  public async getAllDevices(_request: Request, response: Response): Promise<void> {
+    try {
+      const devices: Device[] = await this.deviceService.getDevices()
+      response.send(this.httpResponseFormatter.formatSuccessResponse(devices.map(device => {
+        return this.toDto(device)
+      })))
+    } catch (error) {
+      this.httpErrorHandler.handleError(response, error as Exception)
+    }
+  }
+
+  @GetRequest('/:deviceId')
+  public async getDevice(request: Request, response: Response): Promise<void> {
+    try {
+      const { deviceId } = request.params
+      const device: Device = await this.deviceService.getDevice(deviceId)
+      response.send(this.httpResponseFormatter.formatSuccessResponse(this.toDto(device)))
+    } catch (error) {
+      this.httpErrorHandler.handleError(response, error as Exception)
+    }
+  }
+
+  @PostRequest()
+  public async createDevice(request: Request, response: Response): Promise<void> {
+    try {
+      const device: Device = await request.body
+      if(!Object.values(DeviceType).includes(device.type)){
+        throw new UnprocessableEntityException('device type was not in the correct numerical format')
```

Comment:
This exception is caugt immediatly in this method.
Not sure if that's bad or not...

Also, be nice with some spaces in your if-statement above so it becomes `if (...) {` and not `if(...){`

Reply:
With the refactor to response.status(500).send('Internal Server Error') we no longer need exceptions for this.

My Webstorm setup kills whitespace, but let me see what I can do.